crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -1,4 +1,97 @@
-//! The workflow and queue consumer for sys validation
+//! Holochain workflow to validate all incoming DHT operations with an
+//! app-defined validation function.
+//!
+//! Triggered by system validation, this workflow iterates over a list of
+//! [`DhtOp`]s to be validated, validates each op, updates its validation status
+//! in the database accordingly, and triggers op integration if necessary.
+//!
+//! ### Sequential validation
+//!
+
+// This does not apply right now, because all actions are written to the database
+// straight away in the incoming dht ops workflow. See https://github.com/holochain/holochain/issues/3724
+
+//! Ops are validated in sequence based on their op type and the timestamp they
+//! were authored (see [`OpOrder`] and [`OpNumericalOrder`]). Validating one op
+//! after the other with this ordering ensures that ops that depend on earlier
```

Comment:
```suggestion
//! after the other with this ordering makes it more likely that ops that depend on earlier
```

Reply:
Because we can't actually guarantee that this will happen. Sometimes we will get ops out of order and not be able to include the ops we don't know about in the ordering and end up failing with missing deps then have to wait. That's a normal part of the flow

Reply:
Actually this sentence just says that validating in sequence with a certain order ensures that - validting in sequence with a certain order. I'm rephrasing this.